CodeiumVS/QuickInfo/QuickInfoSource.cs

```diff
@@ -0,0 +1,150 @@
+                text += GetQuickInfoItemText(element);
+            return text;
+        }
+
+        return string.Empty;
+    }
+
+    public async Task<QuickInfoItem> GetQuickInfoItemAsync(IAsyncQuickInfoSession session,CancellationToken cancellationToken)
+    {
+        if (_disposed || session.TextView.TextBuffer != _owner) 
+            return null;
+
+        await GetTagAggregatorAsync(session.TextView);
+
+        Assumes.True(_tagAggregator != null,
+            "Codeium Quick Info Source couldn't create a tag aggregator for error tags");
+
+        // Put together the span over which tags are to be discovered.
+        // This will be the zero-length span at the trigger point of the session.
+        SnapshotPoint? subjectTriggerPoint = session.GetTriggerPoint(_owner.CurrentSnapshot);
+        if (!subjectTriggerPoint.HasValue)
+        {
+            Debug.Fail("The Codeium Quick Info Source is being called when it shouldn't be.");
+            return null;
+        }
+
+        ITextSnapshot currentSnapshot = subjectTriggerPoint.Value.Snapshot;
+        var querySpan = new SnapshotSpan(subjectTriggerPoint.Value, 0);
+
+        // Must be on main thread when dealing with tag aggregator
+        await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+        // Ask for all of the error tags that intersect our query span.  We'll get back a list of mapping tag spans.
+        // The first of these is what we'll use for our quick info.
+        IEnumerable<IMappingTagSpan<IErrorTag>> tags = _tagAggregator.GetTags(querySpan);
+        ITrackingSpan appToSpan = null;
+
+        string problemMessage = string.Empty;
+
+        foreach (var tag in tags.Cast<MappingTagSpan<IErrorTag>>())
+        {
+            NormalizedSnapshotSpanCollection applicableToSpans = tag.Span.GetSpans(currentSnapshot);
+            if ((applicableToSpans.Count == 0) || (tag.Tag.ToolTipContent == null)) continue;
+
+            // We've found a error tag at the right location with a tag span that maps to our subject buffer.
+            appToSpan = currentSnapshot.CreateTrackingSpan(
+                applicableToSpans[0].Span, SpanTrackingMode.EdgeInclusive
+            );
+
+            appToSpan = IntellisenseUtilities.GetEncapsulatingSpan(
+                session.TextView, appToSpan, appToSpan
+            );
+
+            problemMessage += GetQuickInfoItemText(tag.Tag.ToolTipContent) + " and ";
+        }
+
+        if (appToSpan != null && problemMessage.Length > 0)
+        {
+            var hyperLink = ClassifiedTextElement.CreateHyperlink(
+                "Codeium: Explain Problem",
+                "Ask codeium to explain the problem",
+                () => {
+                    ThreadHelper.JoinableTaskFactory.RunAsync(async delegate
+                    {
+                        // TODO: Has the package been loaded at this point?
+                        await CodeiumVSPackage.Instance.LanguageServer.Controller.ExplainProblemAsync(
```

Comment:
There is a chance that the package has not been loaded at this point, so adding a `EnsurePackageLoaded()` here might be necessary.